Group Actions (Invite/Remove/Reject/Accept)

Sumukh opened the pull request:
Frontend forms & Routes for actions. 

This PR also handles CSRF protection for all forms. 
Assignment Dashboard: 
![image](https://cloud.githubusercontent.com/assets/882381/12701402/9fc1b81a-c7bd-11e5-8947-89636e8d3f20.png)

Confirm Removals:
![image](https://cloud.githubusercontent.com/assets/882381/12701418/fb0f4a2a-c7bd-11e5-9bae-d9d2f6eb2a7f.png)

Invite Actions:
![image](https://cloud.githubusercontent.com/assets/882381/12701424/24fdbf4c-c7be-11e5-9abb-0bb6bed7e0b9.png)

Sumukh commented on server/__init__.py:
```diff
@@ -22,6 +24,7 @@
 )
 
 
+
```
Random line? To Remove

Sumukh commented on server/controllers/student.py:
```diff
@@ -89,13 +92,29 @@ def assignment(course, assign):
+    can_remove = False
+    if group:
+        can_invite = len(group.members) < assign.max_group_size
+        can_remove = current_user in [m.user for m in group.members if m.status == 'active']
```
All that's needed here is to figure out if the current user is actually a member. `user_ids` doesn't do this since it always has the current_users id.

knrafto replied:
`can_remove = group.has_status(current_user, 'active')`

Sumukh commented on server/controllers/student.py:
```diff
@@ -165,6 +184,76 @@ def flag(course, assign, bid):
         else:
             assign.unflag(bid, user_ids)
-        return redirect(next_url)
+        if is_safe_redirect_url(request, next_url):
```
Should prevent [open redirects](https://www.owasp.org/index.php/Open_redirect)

Sumukh commented on server/static/css/main.css:
```diff
@@ -17,3 +17,12 @@ pre code  {
+    margin-top: -1em;
+}
+
+td.slim-cell {
```
I don't think this is used anymore in this PR - but I can imagine it being useful later.

knrafto commented on server/static/js/student.js:
```diff
@@ -0,0 +1,19 @@
+        var info = form[0][1].value;
+        swal({
+            title: "Just double checking",
+            text: "Are you sure you want to remove " + info + " from the group?",
```
If you try to leave, it shows "Are you sure you want to remove [your email] from the group?", which could be confusing.

knrafto commented:
The accept/reject buttons are aligned vertically for me:
<img width="894" alt="screen shot 2016-01-31 at 12 28 02 pm" src="https://cloud.githubusercontent.com/assets/2042019/12704741/2a7be130-c816-11e5-9fbd-8abf0cafbcb5.png">

knrafto commented:
For the status column, "in group" and "invited" might be clearer than "active" or "pending" status

knrafto commented:
The accept/reject/remove buttons (that are just icons) should have tooltips explaining what they will do

knrafto commented on server/templates/student/assignment/_tablehelper.html:
```diff
@@ -1,4 +1,6 @@
-{% macro render_subms(subms, course, flagged, tname="Submissions", empty=True) %}
+{% import 'student/assignment/_formhelpers.html' as forms %}
+
+{% macro render_subms(subms, course, flagged, flag_form, tname="Submissions", empty=True) %}
```
`flag_form` is missing when called from the "view all submissions" page

Sumukh commented:
Addressed your comments. 

The copy on the tooltips aren't great - but they should work. Feel free to change them if you have ideas for them. 

I deleted the mocks from the templates
